### DISCUSSION TITLE
xdrip+/libre2/miaomiao1

### DISCUSSION THREAD

**Discussion by evilclivegit** [NightscoutFoundation/xDrip#2010] [Q&A]
Hi -

My sensors have just been changed to libre2, and xdrip hasn't really worked since then.

I'm using a miaomiao 1, and I've upgraded the firmware to 39.
XDrip is 49817a5-2021.10.04
I've got OOP2 70129ac-2022.01.21
Phone is an Honor (cheap huawei) with no NFC, so I don't have librelink. I use a reader most of the time, and xdrip for alarms. Reader is still libre1.

In the event logs I can see data coming in - DexCollectionservice regularly reports "We have all the data that we need 369 checksum_ok = true. But it then doesn't talk to OOP2 to translate that data - very occasionally there is an entry for LibreOOPAlgorithm "handleData called with (etc)", but most of the time nothing.

I'll be honest, I'm a bit lost at this point. Can anybody help me? :-)

cheers,
clive

**Comment by tzachi-dar** [comment]
can you please add "Extra Logging Settings " according to https://www.minimallooper.com/post/how-to-setup-freestyle-libre-2-and-oop2-to-use-a-native-bluetooth-connection-in-xdrip
After that please run for 30 minutes then export your DB and send it to tzachi.dar@gmail.com
I'll try to see what I can understand from this.

**Reply by evilclivegit** [reply]
Thanks. Logging added, and export emailed.

cheers,
clive

**Comment by tzachi-dar** [comment]
Hi,

You need to go to settings->less common settings->other misc options->out of process libre algorithm and disable it.
It should only be used with oop1.

**Reply by evilclivegit** [reply]
But I still need to keep OOP2 running?

**Reply by evilclivegit** [reply]
Anyway, I've unticked that and it's now behaving in a different unhelpful way :-) Still nothing displayed, and interestingly I think it keeps crashing. I'm wondering if the "getData was not able to find a valid time - quiting" log entry under LibreGraph is relevant to that. I'll upload some more logs.

**Comment by tzachi-dar** [comment]
You seem to be doing everything OK, but the program indeed keeps crasshing.
I have added a try-catch with a printing, to see what the issue is.

I have put the new file in your ftp.
You will have to backup your settings and DB, than uninstall and install the new version.
Please run again and send me the logs.

**Reply by evilclivegit** [reply]
For the benefit of others : the version I was given didn't suffer the crashing, and everything is now working. It looks like it was something fixed since the last release.
